bin/analyze

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python3
+
+import matplotlib.pyplot as plt
+from perceval.backends.core.github import GitHub
+from perceval.backends.core.git import Git
+import argparse
+import os
+import sys
+
+from fpdf import FPDF
+import json
+
+from implementations.code_df.conditions import Naive
+
+from implementations.code_df import (
+    code_changes_git,
+    code_changes_lines,
+    )
+from implementations.code_df.utils import str_to_date
+
+GITHUB_URI = "http://github.com/"
+COMMIT_METRICS = \
+            [
+                code_changes_git.CodeChangesGit,
+                code_changes_lines.CodeChangesLines,
+
+            ]
+
+ISSUE_METRICS = []
+
+PULLREQUEST_METRICS = []
+
+METRICS = \
+    {
+        "commit": COMMIT_METRICS,
+        "issue": ISSUE_METRICS,
+        "pull_request": PULLREQUEST_METRICS
+    }
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Analyze script argument parser"
+    )
+
+    parser.add_argument("-a", "--api-token",
+                        help="GitHub API token", default=None)
+
+    parser.add_argument("-r", "--repo",
+                        required=True,
+                        help="GitHub repository, as 'owner/repo'")
+    parser.add_argument("-s", "--since",
+                        default=None,
+                        help="From date")
+    parser.add_argument("-u", "--until",
+                        default=None,
+                        help="To date")
+
+    parser.add_argument("-cat", "--categories",
+                        default=['commit'],
+                        nargs='+',
+                        choices=['commit', 'issue', 'pull_request'],
+                        help="commit, issue, pull_request (any combination)")
+
+    parser.add_argument("-c", "--conditions",
+                        default=[],
+                        nargs='*',
+                        type=list,
+                        help="Commit conds")
+
+    parser.add_argument("-x", "--is-code",
+                        type=list,
+                        default=[Naive()],
+                        help="is code conds")
+
+    parser.add_argument("-t", "--time-series",
+                        action='store_true',
+                        help="To get plots of time-series'")
+
+    parser.add_argument("-pe", "--period",
+                        default='M',
+                        help="period for time-series: 'M', 'W', etc.")
```

Comment:
If the periods are limited, it's probably better to add all of them. WDYT?

Reply:
https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases.
I think they are too many to list in the help section .... WDYT?

Reply:
yes, I agree with you!